### ISSUE TITLE
Breadcrumbs should include pipeline groups

### ISSUE THREAD

**Issue by JohannesRudolph** [concourse/concourse#2322]
# Feature Request

## What challenge are you facing?

So v3.14 added breadcrumbs - yay :+1: 
https://github.com/concourse/concourse/issues/2139 

Unfortunately they feel like a step backwards in terms of usability to me. I'm not sure how it was on v.3.10 but I *think* I had the option of always navigating back to the pipeline group that I came from. 

In any case, my desired UX flow is like: 

- choose pipeline
- choose group
- choose job and unfold task (e.g. to inspect build log)
- go back to group

## A Modest Proposal

Make it easy to go back to the pipeline group I came from - add it as a breadcrumb to the navbar!

* Concourse version: 3.14.1

**Comment by jama-pivotal** [comment]
I think @Lindsayauchin has some designs for this lying around in some of our early prototypes. I think we can dig those out and incorporate it into our current designs

**Comment by chendrix** [comment]
do this please.

**Comment by Lindsayauchin** [comment]
![screen shot 2018-06-29 at 4 08 01 pm](https://user-images.githubusercontent.com/27746614/42112685-a5f87eb2-7bb6-11e8-8e64-d9fe373250cc.png)
![pipeline_breadcrumb_groups](https://user-images.githubusercontent.com/27746614/42112692-aa973e2c-7bb6-11e8-88a7-6b6fb73f10e5.png)

**Comment by JohannesRudolph** [comment]
Hm, does that design imply multi-select of groups for the pipine groups?
Not a bad idea either ;-)

As far as this issue here goes, I'd be happy if the pipeline part as a
whole in the breadcrumb bar (main: develop, workers) would be clickable.

Lindsayauchin <notifications@github.com> schrieb am Fr., 29. Juni 2018,
22:21:

**Comment by jama-pivotal** [comment]
@JohannesRudolph you could have always multi-selected groups by shift+clicking on the group names...even now! https://ci.concourse-ci.org/teams/main/pipelines/main?groups=workers&groups=publish&groups=develop

Or were you hoping for multi-selecting groups _across_ multiple pipelines? that's bit trippy

**Comment by mcwumbly** [comment]
> do this please.

[some additional context](https://github.com/concourse/concourse/issues/2440) on why this would be useful for us.

> Each of these pipelines are large. For example, here's a typical one (of these 41) pipelines that has ~80 jobs split into ~10 pipeline groups:

![ten-groups](https://user-images.githubusercontent.com/2217652/43414081-967cc612-93e6-11e8-9730-78a5ad1e62c0.png)

**Comment by StevenArmstrong** [comment]
When you click on a running job to see the build logs, then click the breadcrumb to navigate back, it doesn't take you to the original pipeline group is proving frustrating for users to navigate. If the pipeline groups could be persisted on the build log view for easy navigation back to a specific pipeline group it would improve the user experience greatly. Having it reset to the first pipeline group is causing lots of complaints.

**Comment by eedwards-sk** [comment]
I have the same issue as @StevenArmstrong 

Since it's dependent upon query params, I would expect it might be possible so that if the user is navigating to a job's build page from a pipeline with groups, then append that group to the breadcrumb.

so e.g. if I have pipeline `foo` with group `bar` and I'm on job `baz`, then on the url:

```
/teams/main/pipelines/foo/jobs/baz/builds/18
```

the breadcrumb will lead to:

```
/teams/main/pipelines/foo?groups=bar
```

**Comment by idavidmcdonald** [comment]
Yep exactly the same issue for us as @StevenArmstrong. It's probably the biggest complaint raised by our team when using concourse as although it's only small, it affects us many times a day. 

@eedwards-sk solution seems reasonable and the least amount of work (rather than say introducing a breadcrumb just for the group). It might require passing the query parameter through so that if you wish to view an individual build you would actually be sent to `/teams/main/pipelines/foo/jobs/baz/builds/18?groups=bar`.

I'm happy to take a shot at implementing this if it seems sensible and would be accepted?

**Comment by JohannesRudolph** [comment]
Not stale. We’d definitely still like to see this. 🙏

**Comment by taylorsilva** [comment]
Adding that this is still not present in v7.12.0

If anyone wants to take a stab at adding it, please do so and make a PR

**Comment by taylorsilva** [comment]
Unless I misunderstand the issue, I think this works now (???). The group doesn't appear in the breadcrumb, but navigating backwards takes you back to the selected group(s) instead of the pipeline's default group.

Video of this on v8.2.4

https://github.com/user-attachments/assets/ec8f0423-5821-4580-9323-bb5f7e158c41

Not sure which PR fixed this, but "hooray" I guess 😅 

I don't plan to re-work the frontend to add groups in the breadcrumbs. If someone wants to do that, feel free to open a PR. I'm closing this as resolved based on the current behaviour I'm seeing.
